ConsumerApi.Tests.Integration/StepDefinitions/ChallengesApiStepDefinitions.cs

```diff
@@ -1,119 +1,109 @@
+            await Identity(identityName).Challenges.CreateChallenge() :
+            await Identity(identityName).Challenges.CreateChallengeUnauthenticated();
+        ChallengeResponse.Should().BeASuccess();
+
+        _challengesContext.ChallengeId = ChallengeResponse.Result!.Id;
+        ChallengeId.Should().NotBeNullOrEmpty();
     }
 
-    [Given("a Challenge c")]
+    [Given(@"a Challenge c")]
     public async Task GivenAChallengeC()
     {
-        var challengeResponse = !_isAuthenticated ? await _sdk.Challenges.CreateChallengeUnauthenticated() : await _sdk.Challenges.CreateChallenge();
-        challengeResponse.Should().BeASuccess();
-        _challengeId = challengeResponse.Result!.Id;
+        _responseContext.ChallengeResponse = _challengesContext.IsAuthenticated ?
+            await AnonymousClient.Challenges.CreateChallenge() :
+            await AnonymousClient.Challenges.CreateChallengeUnauthenticated();
+        ChallengeResponse.Should().BeASuccess();
+
+        _challengesContext.ChallengeId = ChallengeResponse.Result!.Id;
+        ChallengeId.Should().NotBeNullOrEmpty();
+    }
+    #endregion
 
-        _challengeId.Should().NotBeNullOrEmpty();
+    #region When
+    [When("a POST request is sent to the /Challenges endpoint")]
+    public async Task WhenAPostRequestIsSentToTheChallengesEndpoint()
+    {
+        _responseContext.WhenResponse = _responseContext.ChallengeResponse = _challengesContext.IsAuthenticated ?
+            await AnonymousClient.Challenges.CreateChallenge() :
+            await AnonymousClient.Challenges.CreateChallengeUnauthenticated();
     }
 
-    [When("a POST request is sent to the Challenges endpoint")]
-    public async Task WhenAPOSTRequestIsSentToTheChallengesEndpoint()
+    [When(@"(.+) sends a POST request to the /Challenges endpoint")]
+    public async Task WhenISendsAPostRequestToTheChallengesEndpoint(string identityName)
     {
-        _response = _isAuthenticated ? await _sdk.Challenges.CreateChallenge() : await _sdk.Challenges.CreateChallengeUnauthenticated();
+        _responseContext.WhenResponse = _responseContext.ChallengeResponse = _challengesContext.IsAuthenticated ?
+            await Identity(identityName).Challenges.CreateChallenge() :
+            await Identity(identityName).Challenges.CreateChallengeUnauthenticated();
     }
 
-    [When(@"a GET request is sent to the Challenges/{id} endpoint with ""?(.*?)""?")]
-    public async Task WhenAGETRequestIsSentToTheChallengesIdEndpointWith(string id)
+    [When(@"(.+) sends a GET request is sent to the /Challenges/{id} endpoint with ""?(.*?)""?")]
+    public async Task WhenISendsAGetRequestToTheChallengesIdEndpointWith(string identityName, string challengeId)
     {
-        switch (id)
+        switch (challengeId)
         {
             case "c.Id":
-                id = _challengeId;
+                challengeId = _challengesContext.ChallengeId!;
                 break;
             case "a valid Id":
```

Comment:
It's not, but I carried it over since it was already in the tests, and I haven't check all the modules - it might be used there. If I would remove the `"a valid Id"` case, then should I remove the `switch` statement altogether? I would rather not do it, in case of possible future expansion.